accounts/event/event_handler_test.go

```diff
@@ -0,0 +1,135 @@
+package event
+
+import (
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	topic = "testTopic"
+)
+
+type TestEvent struct {
+	Data string
+}
+
+func (te TestEvent) Type() EventType {
+	return TestEventType
+}
+
+func TestSubscribeAndPublish(t *testing.T) {
+	ps := NewEventHandler()
+
+	var err error
+
+	eventChan := make(chan Event, 1)
+
+	ps.Subscribe(topic, eventChan)
+
+	event := TestEvent{Data: "testEvent"}
+
+	ps.Publish(topic, event)
+
+	select {
+	case receivedEvent := <-eventChan:
+		require.Equal(t, event, receivedEvent)
+	case <-time.After(1 * time.Second):
+		err = errors.New("did not receive event")
+	}
+
+	require.NoError(t, err)
+}
+
+func TestUnsubscribe(t *testing.T) {
+	ps := NewEventHandler()
+
+	var err error
+
+	eventChan := make(chan Event, 1)
+
+	ps.Subscribe(topic, eventChan)
+
+	ps.Unsubscribe(topic, eventChan)
+
+	ps.Publish(topic, TestEvent{Data: "testEvent"})
+
+	select {
+	case _, ok := <-eventChan:
+		if ok {
+			err = errors.New("expected channel to be closed")
+		}
+	default:
+	}
+
+	require.NoError(t, err)
```

Comment:
Just as a suggestion in ordet to avoid err variable
```suggestion
var ok bool
	select {
	case _, ok = <-eventChan:
	default:
	}

	require.False(t, ok, "expected channel to be closed")```